src/test/scala/mesosphere/marathon/core/task/tracker/impl/TaskOpProcessorImplTest.scala

```diff
@@ -411,6 +413,114 @@ class TaskOpProcessorImplTest
+      createStateOp = (taskState) => f.stateOpLaunchOnReservation(taskState, Task.Status(stagedAt = f.now)),
+      createStateChange = (taskState) => TaskStateChange.Update(taskState, Some(taskState)))
+  }
+
+  test("MesosUpdate -> Update results processing update steps") {
+    val f = new Fixture
+    verifyStepsAreProcessed(
+      f = new Fixture,
+      createTaskState = (appId) => MarathonTestHelper.mininimalTask(Task.Id.forApp(appId).idString, f.now),
+      createStateOp = (taskState) => f.stateOpUpdate(taskState, MarathonTaskStatusTestHelper.finished, f.now),
+      createStateChange = (taskState) => TaskStateChange.Update(taskState, Some(taskState)))
+  }
+
+  test("MesosUpdate -> Expunge results processing update steps") {
+    val f = new Fixture
+    verifyStepsAreProcessed(
+      f = new Fixture,
+      createTaskState = (appId) => MarathonTestHelper.mininimalTask(Task.Id.forApp(appId).idString, f.now),
+      createStateOp = (taskState) => f.stateOpUpdate(taskState, MarathonTaskStatusTestHelper.finished, f.now),
+      createStateChange = (taskState) => TaskStateChange.Expunge(taskState))
+  }
+
+  test("Expunge -> Expunge results processing update steps") {
+    val f = new Fixture
+    verifyStepsAreProcessed(
+      f = new Fixture,
+      createTaskState = (appId) => MarathonTestHelper.mininimalTask(Task.Id.forApp(appId).idString, f.now),
+      createStateOp = (taskState) => f.stateOpExpunge(taskState),
+      createStateChange = (taskState) => TaskStateChange.Expunge(taskState))
+  }
+
+  test("ReservationTimeout -> Update results processing update steps") {
+    val f = new Fixture
+    verifyStepsAreProcessed(
+      f = new Fixture,
+      createTaskState = (appId) => MarathonTestHelper.mininimalTask(Task.Id.forApp(appId).idString, f.now),
+      createStateOp = (taskState) => f.stateOpReservationTimeout(taskState),
+      createStateChange = (taskState) => TaskStateChange.Update(taskState, oldState = None))
+  }
+
+  test("ReservationTimeout -> Expunge results processing update steps") {
+    val f = new Fixture
+    verifyStepsAreProcessed(
+      f = new Fixture,
+      createTaskState = (appId) => MarathonTestHelper.residentReservedTask(appId),
+      createStateOp = (taskState) => f.stateOpReservationTimeout(taskState),
+      createStateChange = (taskState) => TaskStateChange.Expunge(taskState))
+  }
+
+  test("Reserve -> Update results processing update steps") {
+    val f = new Fixture
+    verifyStepsAreProcessed(
+      f = new Fixture,
+      createTaskState = (appId) => MarathonTestHelper.residentReservedTask(appId),
+      createStateOp = (taskState) => f.stateOpReserve(taskState),
+      createStateChange = (taskState) => TaskStateChange.Update(taskState, oldState = None))
+  }
+
+  def verifyStepsAreProcessed(
+    f: Fixture,
+    createTaskState: (PathId) => Task,
+    createStateOp: (Task) => TaskStateOp,
+    createStateChange: (Task) => TaskStateChange): Unit = {
```

Comment:
Subjective:

I think that I (personally) would have understood the test better if the parameters were called

* initialTaskState
* stateOp
* expectedStateChange

The "create" prefix would not be necessary for me. A diferrentiation between expected and given state would help.